src/OrchardCore/OrchardCore.Abstractions/Localization/CultureScope.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Globalization;
+
+namespace OrchardCore.Localization
+{
+    public sealed class CultureScope : IDisposable
+    {
+        private readonly CultureInfo _originalCulture;
+        private readonly CultureInfo _originalUICulture;
+
+        private CultureScope(CultureInfo culture, CultureInfo uiCulture)
+        {
+            Culture = culture;
+            UICulture = uiCulture;
+            _originalCulture = CultureInfo.CurrentCulture;
+            _originalUICulture = CultureInfo.CurrentUICulture;
+
+            SetCultures(culture, uiCulture);
+        }
+
+        public CultureInfo Culture { get; }
+
+        public CultureInfo UICulture { get; }
```

Comment:
@hishamco i will approve your PR

But just one thing, i don't see any usage of these 2 public properties

Reply:
I point to where it's used, also we might to expose the original culture because may it's useful in some cases